apps/class-solid/src/components/EditableText.tsx

```diff
@@ -0,0 +1,47 @@
+import { Show, createSignal } from "solid-js";
+import { cn } from "~/lib/utils";
+import { Button } from "./ui/button";
+
+export function EditableText(props: {
+  text: string;
+  onChange: (text: string) => void;
+  class?: string;
+}) {
+  const [editing, setEditing] = createSignal(false);
+  const fallback = (
+    <button
+      type="button"
+      class={cn(props.class, "group")}
+      title="Click to change"
+      onClick={(e) => {
+        // e.preventDefault();
+        setEditing(true);
+      }}
+    >
+      {props.text}
+      <span class="invisible ps-1 group-hover:visible">🖉</span>
+    </button>
+  );
+  return (
+    <Show when={editing()} fallback={fallback}>
+      <form
+        onSubmit={(event) => {
+          event.preventDefault();
+          const formData = new FormData(event.currentTarget);
+          props.onChange(formData.get("text") as string);
+          setEditing(false);
```

Comment:
This is nice, but I think `setEditing` should also be set to false when the user clicks outside the form, or presses escape, or changes focus otherwise. 

Also, would it make sense to re-use our solidUI textifield here?

Reply:
See react example at https://github.com/i-VRESSE/haddock3-webapp/blob/c1a6ef1f2d7939ff722235df7da08dcf3e8feb13/app/components/JobName.tsx#L35-L41 for escape or blur and lower how to focus textfield.

I tried to use TextField but it complained about not being inside a formcontext

Reply:
To cancel on loosing focus, you should listen to focusout event on form like

```jsx
<form
...
onFocusOut={(e) => {
          if (!e.currentTarget.contains(e.relatedTarget as Node)) {
             setEditing(false);
          } else {
            // Focus within form
          }
        }}
/>
```

Reply:
implemented